source/StartConditions.cpp

```diff
@@ -292,7 +292,7 @@ const string &StartConditions::GetSystemName() const noexcept
 const string &StartConditions::GetDateString() const noexcept
 {
 	auto it = infoByState.find(state);
-	return it == infoByState.end() ? ILLEGAL : it->second.date;
+	return it == infoByState.end() ? ILLEGAL : it->second.date ? it->second.date.ToString() : it->second.dateString;
```

Comment:
This isn't great to read.
```suggestion
	if(it == infoByState.end())
		return ILLEGAL;
	if(it->second.date)
		return it->second.date.ToString();
	return it->second.dateString;
```